shared/performance/performance.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package performance
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/knative/test-infra/shared/junit"
+	"github.com/knative/test-infra/shared/mysql"
+)
+
+const (
+	presubmit  = "presubmit"
+	dbName     = "knative_performance"
+	dbInstance = "knative-tests:us-central1:knative-monitoring"
+	insertStmt = `
+	INSERT INTO METRICS (
+		RunId, JobType, TestName, MetricName, MetricValue
+		) VALUES (?, ?, ?, ?, ?)`
+
+	// Path to secrets for username and password
+	userSecret = "/secrets/cloudsql/monitoringdb/username"
+	passSecret = "/secrets/cloudsql/monitoringdb/password"
+
+	// Property name used by testgrid
+	perfLatency = "perf_latency"
+)
+
+// CreatePerfTestCase creates a perf test case with the provided name and value
+func CreatePerfTestCase(metricValue float32, metricName, testName string) junit.TestCase {
+	tp := []junit.TestProperty{{Name: perfLatency, Value: fmt.Sprintf("%f", metricValue)}}
+	tc := junit.TestCase{
+		ClassName:  testName,
+		Name:       fmt.Sprintf("%s/%s", testName, metricName),
+		Properties: junit.TestProperties{Properties: tp}}
+	return tc
+}
+
+type DBConfig struct {
+	mysql.DBConfig
+}
+
+// Configure the db instance to store metrics information.
+// This will be later used to show the trending metrics on our grafana dashboard.
+func ConfigureDB() (*DBConfig, error) {
```

Comment:
I think this is also defined in monitor/main.go. Move it to separate file (maybe `shared/mysql/config.go`?)

Reply:
yes. but the dbname here is different and I feel every module should be able to define their own secrets and db names and does not make sense in common

Reply:
I think we can take secret file and db name, db instance as params for this function. I think it still make sense to move it to common since the underlying logic is identical.

Reply:
agreed. done.